### PR TITLE
BUG: Verifying that directory exists instead of java executable

### PR DESCRIPTION
**Pull request by fbudin69500**
To assess the directory in which the 'java' executable is installed,
SCIFIO was verifying that the given directory existed. However,
in certain cases, the directory existed, but the 'java' executable
did not exist. This could easily happened with the install directory
which by default is set to '/usr/local/' and that is likely to
exist. This would result in SCIFIO trying to use that path
to find the 'java' executable instead of searching in the PATH.
This bug is more easily visible on MacOS as SCIFIO assumes that
java should be installed on the system whereas for Windows and Linux
java is downloaded.

### CODE REVIEW

**Comment by ctrueden** [comment]
Thanks @fbudin69500. Is there anyone else at Kitware who would care to test this out, before we merge?

**Comment by fbudin69500** [comment]
@thewtex Do you want to test my commit?

File: src/itkSCIFIOImageIO.cxx.in [hunk 1]
```diff
@@ -255,6 +255,28 @@ std::string SCIFIOImageIO::FindDimensionOrder(const ImageIORegion & region)
   return command;
 }
 
+bool SCIFIOImageIO::CheckJavaPath(std::string javaHome, std::string &javaCmd)
+{
+  std::vector<std::string> javaCmdPath;
+#if defined(WIN32)
```
Comment (thewtex) [inline src/itkSCIFIOImageIO.cxx.in]: Please use `_WIN32` because `WIN32` is not defined in all cases on Windows.

**Comment by thewtex** [comment]
@fbudin69500 Yes, will test on an OSX system.

Comment made inline for a suggested unrelated fix.

**Comment by thewtex** [comment]
Build without warnings and tests pass on the misty box. :+1:

**Comment by fbudin69500** [comment]
I will address @thewtex's comment and amend my PR.

**Comment by fbudin69500** [comment]
PR amended. Let me know if you want me to do anything else before this can be merged. Thanks!

**Comment by thewtex** [comment]
Thanks!

**Pull request merged**